<documentation title="Restricted Date and Time Functions">
    <standard>
        <![CDATA[
      The restricted functions date_default_timezone_set() and date() should not be used.

      Using date_default_timezone_set() isn't allowed, because WordPress core needs default time zone set to UTC. Use DateTime object or WP core API functions instead.

      Using date() isn't allowed, as it is affected by runtime timezone changes which can cause date/time to be incorrectly displayed. Use gmdate() instead.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Using DateTime object.">
            <![CDATA[
$date = new DateTime();
$date->setTimezone(
    new DateTimeZone( 'Europe/Amsterdam' )
);
        ]]>
        </code>
        <code title="Invalid: Using date_default_timezone_set().">
            <![CDATA[
date_default_timezone_set( 'Europe/Amsterdam' );
        ]]>
        </code>
    </code_comparison>
    <code_comparison>
        <code title="Valid: Using gmdate().">
            <![CDATA[
$last_updated = gmdate(
    'Y-m-d\TH:i:s',
    strtotime( $plugin['last_updated'] )
);
        ]]>
        </code>
        <code title="Invalid: Using date().">
            <![CDATA[
$last_updated = date(
    'Y-m-d\TH:i:s',
    strtotime( $plugin['last_updated'] )
);
        ]]>
        </code>
    </code_comparison>
</documentation>
